crates/sui-graphql-rpc/src/server/data_provider.rs

```diff
@@ -216,6 +221,108 @@ fn convert_bal(b: sui_json_rpc_types::Balance) -> Balance {
         total_balance: BigInt::from_str(&format!("{}", b.total_balance)).unwrap(),
     }
 }
+pub(crate) async fn fetch_gas_input(cl: &SuiClient, digest: &String) -> Result<GasInput> {
+    let tx_block = cl
+        .read_api()
+        .get_transaction_with_options(
+            TransactionDigest::from_str(digest)?,
+            SuiTransactionBlockResponseOptions::full_content(),
+        )
+        .await?;
+    let tx = tx_block.transaction.unwrap().data;
+    let gas_data = tx.gas_data();
+
+    let payment_obj_ids = gas_data
+        .payment
+        .clone()
+        .iter()
+        .map(|x| x.object_id)
+        .collect::<Vec<_>>();
+
+    let obj_responses: Vec<SuiObjectResponse> =
+        future::try_join_all(payment_obj_ids.iter().map(|x| {
+            cl.read_api()
+                .get_object_with_options(*x, SuiObjectDataOptions::full_content())
```

Comment:
The multi-get API seems like a better fit here?

Reply:
Not for this PR, but as a general design point we should think about an optimisation where if we spot that this field is not requested on gas data, or only the ID of the payment objects is requested, we avoid making this fetch.